src/tactic/norm_swap.lean

```diff
@@ -0,0 +1,79 @@
+/-
+Copyright (c) 2021 Yakov Pechersky All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yakov Pechersky
+-/
+import data.equiv.basic
+import tactic.norm_num
+
+/-!
+# `norm_swap`
+
+Evaluating swapping expressions of numerals, of the form `swap x y z`,
+for numerals `x y z : ℕ`.
+-/
+
+open equiv tactic expr
+
+/--
+If `e : expr` is of the form `app (⇑f) x`, get back the `expr`s for type of `f`,
```

Comment:
I think this formulation might be more straightforward: 

```lean
/--
Match a term of shape `app (⇑f) x` and return:
  * the type `tf` of `f`
  * the instance `has_coe_to_fun tf`
  * `f`
  * `x`
  -/

Reply:
In general, I see that you use phrasing like "the `expr` which is the *something something*". I think you could stick to saying "the *something something*" adding parenthetical notes if there's any possible ambiguity.